### PR TITLE
resolve red builds on the develop branch after VS2017 15.3

### PR DESCRIPTION
**Pull request by olevett**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Fix for #1420

**What is the current behavior? (You can also link to an open issue here)**
Appveyor is old version

**What is the new behavior (if this is a feature change)?**
Appveyor is preview builds

**What might this PR break?**
Hopefully nothing

**Please check if the PR fulfills these requirements**
- [ ] The commit follows our guidelines: https://github.com/reactiveui/reactiveui#contribute
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Other information**:
Have raised a ticket on Cake, as the VSWhere tool we're using doesn't currently include pre-release builds, so VS can't find what it's doing. The build is falling back to MSBuild 14, which doesn't support new csprojs, hence breaking the build.

Cake ticket https://github.com/cake-build/cake/issues/1754, notes on vswhere prerelease flag https://blogs.msdn.microsoft.com/heaths/2017/06/24/vswhere-version-2-0-released/

### CODE REVIEW

**Comment by olevett** [comment]
Note, this shouldn't be merged till Cake supports pre-release

**Comment by ghuntley** [comment]
I've hijacked this PR to get us to green builds on the develop branch. 

Changes:
- No longer manually installed the android sdk as appveyour is now up to date (this speeds up our build times as well)
- Moved back to non-preview builds as appveyor has updated.
- Fix for #1420
- Fix for #1409

File: build.cake [hunk 1]
```diff
@@ -200,14 +206,13 @@ Task("BuildReactiveUI")
...
+        Information("Building {0} using {1}", solution, msBuildPath);
 
         MSBuild(solution, new MSBuildSettings() {
-                ToolPath= msBuildPath
+                ToolPath = msBuildPath,
+                ArgumentCustomization = args => args.Append("/bl:reactiveui.binlog")
             }
-            .WithTarget("restore;pack")
+            .WithTarget("restore;build;pack")
             .WithProperty("PackageOutputPath",  MakeAbsolute(Directory(artifactDirectory)).ToString())
```
Comment (ghuntley) [inline build.cake]: Had to add `build` here to generate our unit test assemblies (`ReactiveUI.Tests/bin/**/`). Before VS2017 15.3 they were generated automatically via the pack step.

File: build.cake [hunk 2]
```diff
@@ -122,10 +124,14 @@ Task("BuildEventBuilder")
     .Does (() =>
 {
     var solution = "./src/EventBuilder.sln";
+    Information("Building {0} using {1}", solution, msBuildPath);
 
     NuGetRestore(solution, new NuGetRestoreSettings() { ConfigFile = "./src/.nuget/NuGet.config" });
 
-    MSBuild(solution, new MSBuildSettings()
+    MSBuild(solution, new MSBuildSettings() {
+            ToolPath = msBuildPath,
```
Comment (ghuntley) [inline build.cake]: Had to specify the location of msbuild (an oversight, we do this as part of the `buildreactiveui` step and forgot to add it in here)

File: src/Directory.build.props [hunk 3]
```diff
@@ -18,7 +18,7 @@
   </PropertyGroup>
   
   <ItemGroup>
-    <PackageReference Include="MSBuild.Sdk.Extras" Version="1.0.1" PrivateAssets="All" />
+    <PackageReference Include="MSBuild.Sdk.Extras" Version="1.0.6" PrivateAssets="All" />
   </ItemGroup>
```
Comment (ghuntley) [inline src/Directory.build.props]: dis got bumped; it's very important but I don't have the backstory.

File: build.cake (hunk 1 above)
Comment (rohit21agrawal) [inline build.cake]: it could be that ```IsPackable``` has been set to false in Test project sdk targets , and that's why pack doesn't do anything here.

**Pull request merged**